pkg/cable/libreswan/libreswan.go

```
@@ -77,7 +77,11 @@ const defaultNATTPort = "4500"
 const ipsecSpecEnvVarPrefix = "ce_ipsec"
 
 // NewLibreswan starts an IKE daemon using Libreswan and configures it to manage Submariner's endpoints
-func NewLibreswan(localEndpoint types.SubmarinerEndpoint, localCluster types.SubmarinerCluster) (cable.Driver, error) {
+func NewLibreswan(localEndpoint *types.SubmarinerEndpoint, localCluster *types.SubmarinerCluster) (cable.Driver, error) {
+	if localEndpoint == nil || localCluster == nil {
+		return nil, fmt.Errorf("NewLibreswan called with nil localEndpoint and/or localCluster")
+	}
```

I think it's reasonable to expect these to be non-nil, ie part of the API contract with the caller, and it would be a programmatic error if nil so just let it panic. Plus it adds another code path for UTs to cover that should never happen.